### PR TITLE
MULE-19355: Munit Runtime Validations

### PR DESCRIPTION
**Pull request by pabloperalta**

### CODE REVIEW

**Comment by pabloperalta** [comment]
--validate

File: modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    ArtifactClassLoaderModelBuilder artifactClassLoaderModelBuilder = new ArtifactClassLoaderModelBuilder(artifactFolder, artifactBundleDescriptor) {
+      @Override
+      protected List<URI> processPluginAdditionalDependenciesURIs(BundleDependency bundleDependency) {
+        return null;
+      }
+
+      @Override
+      protected List<String> getActiveProfiles() {
+        return Lists.newArrayList(profileId);
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java:46]: Use `Arrays.asList()` or `Collections.singletonList()`

Comment (elrodro83) [inline modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java:46]: static import

File: modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      }
+    };
+
+    // When
+    try {
+      artifactClassLoaderModelBuilder.findArtifactPackagerPlugin(model);
+    }catch (Throwable t){
+      fail();
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java:54]: this fail would hide tha actual reason the test was in error

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java (hunk 1 above)
Comment (pabloperalta) [reply modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java]: Done

Comment (pabloperalta) [reply modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java]: Done

File: modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java (hunk 2 above)
Comment (pabloperalta) [reply modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/maven/ArtifactClassLoaderModelBuilderTestCase.java]: Changed the catch for NPE only and added a message to the fail() method. Any other exception will bubble up unchatched

**Review by elrodro83 — APPROVED** [review]

**Review by elrodro83 — APPROVED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**